### PR TITLE
 Merge only works when trigger phrase is present #257 

### PR DESCRIPTION
**Pull request by dulvac**
Adds a new checkbox parameter to the merger post build action that allows merging without a trigger phrase if configured like that.

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/ghprb/jobdsl/GhprbPullRequestMergeContext.java [hunk 1]
```diff
@@ -71,4 +72,19 @@ public void deleteOnMerge(boolean deleteOnMerge) {
     public void deleteOnMerge() {
         deleteOnMerge(true);
     }
+
+    /**
+     * Allows merging the PR even if the trigger phrase was not present. Dewfaults to {@code false}
```
Comment (spredfast-tugboat) [inline src/main/java/org/jenkinsci/plugins/ghprb/jobdsl/GhprbPullRequestMergeContext.java]: Typo at "Defaults"

Comment (bchavet) [inline src/main/java/org/jenkinsci/plugins/ghprb/jobdsl/GhprbPullRequestMergeContext.java]: Typo at "Defaults"

File: src/main/java/org/jenkinsci/plugins/ghprb/jobdsl/GhprbPullRequestMergeContext.java [hunk 2]
```diff
@@ -71,4 +72,19 @@ public void deleteOnMerge(boolean deleteOnMerge) {
...
+     * @param allowMergeWithoutTriggerPhrase
+     */
+    public void allowMergeWithoutTriggerPhrase(boolean allowMergeWithoutTriggerPhrase) {
+        this.allowMergeWithoutTriggerPhrase = allowMergeWithoutTriggerPhrase;
+    }
+
+    /**
+     * Allows merging the PR even if the trigger phrase was not present. Dewfaults to {@code false}
```
Comment (bchavet) [inline src/main/java/org/jenkinsci/plugins/ghprb/jobdsl/GhprbPullRequestMergeContext.java]: Typo at "Defaults"

File: pom.xml [hunk 3]
```diff
@@ -10,7 +10,7 @@
 
     <artifactId>ghprb</artifactId>
     <name>GitHub Pull Request Builder</name>
-    <version>1.32.9-SNAPSHOT</version>
+    <version>1.32.9-dulvac-SNAPSHOT</version>
```
Comment (DavidTanner) [inline pom.xml]: Please revert your pom.xml changes.

**Comment by DavidTanner** [comment]
Other than the pom.xml changes I think everything looks good.

File: pom.xml (hunk 3 above)
Comment (dulvac) [inline pom.xml]: Ooups, terribly sorry about that :/

**Comment by dulvac** [comment]
retest please

**Comment by dulvac** [comment]
test this please

**Comment by dulvac** [comment]
@DavidTanner , your jenkins job seems to time out after 30 minutes. Slow download?

**Comment by DavidTanner** [comment]
I am not sure what is going on.  That job is the main build that the Jenkins team set up, I have no control over it

File: pom.xml [hunk 4]
```diff
@@ -28,15 +28,15 @@
     </developers>
 
     <scm>
-        <connection>scm:git:ssh://github.com/jenkinsci/ghprb-plugin.git</connection>
-        <developerConnection>scm:git:ssh://git@github.com/jenkinsci/ghprb-plugin.git</developerConnection>
-        <url>https://github.com/jenkinsci/ghprb-plugin</url>
+        <connection>scm:git:ssh://github.com/dulvac/ghprb-plugin.git</connection>
```
Comment (DavidTanner) [inline pom.xml]: All changes to pom.xml need to be reverted here.

Comment (dulvac) [inline pom.xml]: done

**Comment by dulvac** [comment]
well... seems to work now

**Comment by dulvac** [comment]
@DavidTanner Anything else I can do?

**Comment by dulvac** [comment]
Hi guys, It is possible to merge this into master?

**Comment by brettneese** [comment]
+1, our team could really use this

**Pull request merged**